cpp/include/rapidsmpf/bootstrap/utils.hpp

```diff
@@ -0,0 +1,124 @@
+/**
+ * SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include <rapidsmpf/bootstrap/types.hpp>
+
+namespace rapidsmpf::bootstrap {
+
+/**
+ * @brief Get current CPU affinity as a string.
+ *
+ * Queries the current process's CPU affinity mask and formats it as a
+ * comma-separated list of CPU core IDs, with ranges represented as "start-end".
+ *
+ * Example output: "0-3,8-11" for cores 0,1,2,3,8,9,10,11
+ *
+ * @return CPU affinity string, or empty string on error.
+ */
+std::string get_current_cpu_affinity();
+
+/**
+ * @brief Get current NUMA node(s) for memory binding.
+ *
+ * Queries the NUMA node associated with the CPU the current process is running on.
+ * This is a best-effort approach and may not be accurate in all cases.
+ *
+ * @return Vector of NUMA node IDs. Empty if NUMA is not available or detection fails.
+ */
+std::vector<int> get_current_numa_nodes();
```

Comment:
I think it would be good to move them into `<rapidsmpf/utils.hpp>`, or possibly into a new header such as `system_info.hpp`. They will also be useful for `PinnedMemoryResource`, for example.